eth_keys/datatypes.py

```diff
@@ -381,6 +340,15 @@ def __getitem__(self, index: int) -> int:  # type: ignore
     def __repr__(self) -> str:
         return "'{0}'".format(self.to_hex())
 
+    def __index__(self) -> int:
+        return self.__int__()
```

Comment:
@carver see: https://docs.python.org/3/library/operator.html#operator.__index__

It's part the python API for interpreting values as integers.

Reply:
Yeah, I was basing the comment on this history: https://docs.python.org/2.5/whatsnew/pep-357.html

So AFAICT, `obj.__index__()` is only called when you do something like `mylist[obj:]`

Reply:
I think it was just a blind inclusion by myself a long time ago.  @jannikluhn can you try dropping this method and see if anything in the tests breaks?

Reply:
Oooh, I missed that it was there before.

Reply:
Removing it breaks this test: https://github.com/ethereum/eth-keys/blob/master/tests/core/test_key_and_signature_datastructures.py#L115

Looks like `__index__` is used for `hex`, instead of `__hex__`. To me this was surprising. but apparently `__hex__` got removed with python 3 (I can't find it in the docs anymore at least).

In my opinion, `hex` support should probably be removed altogether as it doesn't guarantee a length, so `hex(signature)` is not always the same as `signature.to_hex()`. But maybe in a separate PR?

Reply:
> But maybe in a separate PR?

:+1: 

Can you open an issue for this.